USAGE.md

```diff
@@ -0,0 +1,125 @@
+Example Usage
+=============
+
+Not entirely aligned with current `allocator_common.h`, just wanted to try
+and capture some thoughts about how the usage of this API might roughly look.
+
+Note that we want to take into account scenarios where some processes may not
+be permitted to open all devices.  For example a client of the display server
+running in a sandbox, etc.  And we don't want window-system specific IPC
+knowledge in "liballoc".
+
+So each `alloc_dev_t` should be exposed to the user, and logic for
+constructing the union of constraints and intersection of capabilities
+should be decoupled from `alloc_dev_t`.  And of course, constraints/
+capabilities should be easily serializable.
+
+To keep things simple, a hypothetical example with a camera + display, in
+single process but with some comments about where things would be serialized
+in a multi-process scenario.
+
+All error checking omitted for brevity.
+
+Open Devices
+------------
+
+Generally an `alloc_dev_t` represents a single device file or API.  In cases
+where the underlying device file is not exposed (such as OpenMAX or OpenCL,
+or OpenGL when not running on top of gbm) an extension would be required to
+retrieve the `alloc_dev_t` from that particular API.
+
+```
+alloc_dev_t *display = alloc_dev_create(open("/dev/dri/card0", O_RDWR));
+alloc_dev_t *camera  = alloc_dev_create(open("/dev/video0", O_RDWR));
```

Comment:
Well, I think in some cases, a device fd is all you have (such as v4l), since there is no userspace API.  In other cases where userspace doesn't already have a device fd (like OpenGL app on x11), some API specific extension would be needed to retrieve the `alloc_dev_t`.

I guess an open question on how that would work for the "bare metal" GL app, like a weston compositor.  I guess that gets down to the question of whether this new API replaces or complements gbm (issue #1).

I hadn't really considered UUIDs/LUIDs.. maybe that could work.  But fd's just seemed convenient.  Otherwise you get into the question of how to identify different devices.. for example you might have a bunch of different /dev/videoN v4l devices (front/back camera, video encoder, decoder, etc).